### PR TITLE
Unvanquished: new flatpak

### PR DESCRIPTION
**Pull request by necessarily-equal**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. Note I do not have commit rights. **Link:** https://github.com/Unvanquished/
- [x] Upstream owns the domain used in the application ID. https://unvanquished.net/
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

The patch isn't submitted upstream. We are discussing what the proper semantics we should prefer (probably removing the -connect?) to have a *single* desktop file. We don't have the requirement outside of flatpaks, because we do simply distribute two of them. https://github.com/Unvanquished/Unvanquished/issues/1424

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

CC @illwieckz

### CODE REVIEW

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: net.unvanquished.Unvanquished.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      - type: file
+        path: net.unvanquished.Unvanquished.appstream.xml
+      - type: file
+        path: net.unvanquished.Unvanquished.desktop
+      - type: archive
+        dest: external_deps/linux64-5
+        url: https://dl.unvanquished.net/deps/linux64-5.tar.bz2
+        sha256: 0dc5a58403926b0fb4680fb64220cc10b58c83e8871e7c59285ff5161287eb17
+      - type: patch
+        path: 0001-Suppress-a-warning.patch
```
[the remark this answers is not in the document]
Comment (hfiguiere) [reply net.unvanquished.Unvanquished.yml:58]: nevermind. I also missed it.

**Comment by hfiguiere** [comment]
bot, build net.unvanquished.Unvanquished

File: net.unvanquished.Unvanquished.desktop [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Version=1.5
+Name=Unvanquished
+Comment=FPS/RTS Game - Aliens vs. Humans
+Type=Application
+Terminal=false
+Exec=/app/bin/unvanquished -connect %u
```
[the remark this answers is not in the document]
Comment (necessarily-equal) [reply net.unvanquished.Unvanquished.desktop:7]: Is this safe from any conflict with the outside? For example if people would download a distro package too?

**Comment by necessarily-equal** [comment]
Should build fine now

**Comment by hfiguiere** [comment]
bot, build net.unvanquished.Unvanquished

File: net.unvanquished.Unvanquished.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      - >
+        for d in unv/dist/icons/*; do
+          mkdir /app/share/icons/hicolor/$(basename $d)/apps/ -p
+          install -Dm0644 $d/unvanquished.png /app/share/icons/hicolor/$(basename $d)/apps/net.unvanquished.Unvanquished.png
+        done
+
+  - name: unvanquished
+    buildsystem: simple
+    build-commands:
+      - cp -a unvanquished /app/bin/
```
Comment (hfiguiere) [inline net.unvanquished.Unvanquished.yml:77]: There is no `/app/bin` so you need to create it. Usually we use `install` instead of `cp`

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
Have the appdata and destkop files submitted upstream?

**Comment by hfiguiere** [comment]
bot, build net.unvanquished.Unvanquished

**Comment by necessarily-equal** [comment]
The desktop file is a tweaked upstream one. It was initially sed'ed but at the 4th sed command I thought it was simpler to copy it. The appstream is in-PR, https://github.com/Unvanquished/Unvanquished/pull/1422, but the chicken-and-egg problem on release time is annoying. Not sure what the usual way to handle it is.

**Comment by hfiguiere** [comment]
> The desktop file is a tweaked upstream one. It was initially sed'ed but at the 4th sed command I thought it was simpler to copy it. The appstream is in-PR, [Unvanquished/Unvanquished#1422](https://github.com/Unvanquished/Unvanquished/pull/1422), but the chicken-and-egg problem on release time is annoying. Not sure what the usual way to handle it is.

It's better to patch then.

**Comment by necessarily-equal** [comment]
I must say I have no idea why the build fails

**Comment by hfiguiere** [comment]
bot, build net.unvanquished.Unvanquished

**Comment by hfiguiere** [comment]
tbf not sure what's happening here either.

**Comment by necessarily-equal** [comment]
looks like it tries to fetch Unvanquished.git/src/utils/cbse ; but this repo shouldn't be recursed

```yaml
      - type: git # used for the icons
        disable-submodules: true
        dest: unv 
        url: https://github.com/Unvanquished/Unvanquished
        tag: unvanquished/0.52.0
```

**Comment by necessarily-equal** [comment]
about the desktop file, I've opened https://github.com/Unvanquished/Unvanquished/issues/1424 and https://github.com/Unvanquished/Unvanquished/issues/1423. I don't really want to clone yet another repo to get one half of a desktop file and add the other half manually. So I'd keep it like that for now hoping to get it right on the next release.

**Comment by necessarily-equal** [comment]
About the build error, well, locally it works :shrug:

**Comment by hfiguiere** [comment]
> about the desktop file, I've opened [Unvanquished/Unvanquished#1424](https://github.com/Unvanquished/Unvanquished/issues/1424) and [Unvanquished/Unvanquished#1423](https://github.com/Unvanquished/Unvanquished/issues/1423). I don't really want to clone yet another repo to get one half of a desktop file and add the other half manually. So I'd keep it like that for now hoping to get it right on the next release.

You don't need to clone repo.  All you need is to add a patch. Github allow getting individual patches for commit.

**Comment by necessarily-equal** [comment]
> You don't need to clone repo. All you need is to add a patch. Github 
> allow getting individual patches for commit.
> 

Well, the desktop files are stored in the updater repo, 
https://github.com/Unvanquished/updater/

**Comment by barthalion** [comment]
bot, build net.unvanquished.Unvanquished

**Comment by barthalion** [comment]
bot, build net.unvanquished.Unvanquished

**Comment by barthalion** [comment]
bot, build net.unvanquished.Unvanquished

**Comment by necessarily-equal** [comment]
Any reason for dropping the thumbnails, may I ask?

File: net.unvanquished.Unvanquished.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      - -DUSE_CURSES=FALSE # broken, not sure why
+      - -DUSE_GEOIP=FALSE # this is mostly useful for server anyway
+      - -DUSE_HARDENING=TRUE
+      - -DUSE_LTO=TRUE
+    sources:
+      - type: git
+        url: https://github.com/DaemonEngine/Daemon
+        tag: unvanquished/0.52.0
+      - type: file
+        path: net.unvanquished.Unvanquished.metadata.xml
```
Comment (hfiguiere) [inline net.unvanquished.Unvanquished.yml:44]:
```suggestion
        path: net.unvanquished.Unvanquished.metainfo.xml
```

Comment (barthalion) [reply net.unvanquished.Unvanquished.yml:44]: Bah.

**Comment by barthalion** [comment]
bot, build net.unvanquished.Unvanquished

**Comment by barthalion** [comment]
OK, it's something:

```
builddir/files/share/appdata/net.unvanquished.Unvanquished.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
```

@necessarily-equal Mind adding OARS section?

Re: thumbnails, appstream-util generates thumbnails from the source images and then we serve these from CDN, so it would be overwritten anyway.

**Comment by necessarily-equal** [comment]
> @necessarily-equal Mind adding OARS section?

I did write one.

I have some doubts about some sections though. For example the discrimination thingy: all members of a team are equal and it's fine, but the objective of the game is to eliminate the other team. You can call it pest-control, but you could also consider it to be a genocide? Likewise, human-to-alien is fantasy violence (non-existing target) with realistic weapons (guns), but alien-to-human is against real-looking persons (cloned humans), but with unrealistic weapons. Since none of this is possible in real life, I did put mild — but moderate could be appropriate too

**Comment by necessarily-equal** [comment]
Also in-game chat can be moderated, but we can't make sure every server is. Should I put *intense* here?

**Comment by barthalion** [comment]
Eh, in reality it's just the best effort rating. What you picked looks sane to me.

bot, build net.unvanquished.Unvanquished

**Comment by barthalion** [comment]
It displays following messages for me (in the terminal, but the last one also in UI):

```
Warn: Could not open an OpenAL device. 
Warn: Couldn't initialize the audio subsystem
```

**Comment by hfiguiere** [comment]
Maybe it needs `--device=all` for audio (not using pulse). Just an idea.

**Comment by necessarily-equal** [comment]
I've force-disabled the openAL backend, does that work for you?

**Comment by illwieckz** [comment]
When installing this flatpak on a computer running Nvidia's 340 driver, flatpak installs `org.freedesktop.Platform.GL.nvidia-340-108` and then the `Warn: GLW_StartOpenGL() - could not load OpenGL subsystem: Missing GL version` is error is faced. see https://github.com/DaemonEngine/Daemon/issues/474

One known fix is to add `-DOpenGL_GL_PREFERENCE=LEGACY` to cmake call. This means the game will probably not be `GLVND` compatible, but the game would be expected to run everywhere there is a working GL driver.

Note: the prebuilt binary from `unvanquished_0.52.0.zip` is expected to not be affected.

**Comment by necessarily-equal** [comment]
I've added the cmake flags. I'll add it to the nixpkg too, just in case.

**Comment by AsciiWolf** [comment]
Sound works fine now. Tested on my Fedora 33 system. Also, I did not notice any issues, game seems to be fully playable. :-)

**Comment by AsciiWolf** [comment]
I have tried enabling OpenAL with `-set s_useOpenAL=1` and sound still works fine. @barthalion What distro you are trying it on? Fedora 34? Maybe it is a PipeWire related bug.

**Comment by AsciiWolf** [comment]
@necessarily-equal Why did you close the PR?

**Comment by necessarily-equal** [comment]
I've got no idea why github closed the PR — I wasn't even connected. Can someone reopen it?

**Comment by necessarily-equal** [comment]
About the openAL error, forget about `s_useOpenAL`, it doesn't do anything

I have seen this warning form time to time. Maybe it is related?
```
--- Common Initialization Complete --- 
Calling GetRefAPI… 
SDL_Init( SDL_INIT_VIDEO )...  
Using SDL Version 2.0.12 
]
** (process:181058): WARNING **: 13:54:45.130: Error writing credentials to socket: Error sending message: Broken pipe
SDL using driver "x11" 
Initializing OpenGL display 
```

It seems to be a race condition. Not sure why it happens.

**Comment by barthalion** [comment]
I think only you can re-open it but I don't think it really matters.

@AsciiWolf yeah, F34.

**Comment by barthalion** [comment]
bot, build net.unvanquished.Unvanquished

**Comment by barthalion** [comment]
It works for me now. Thanks @necessarily-equal for sorting it out.

/merge @illwieckz

**Comment by AsciiWolf** [comment]
@necessarily-equal @barthalion Any chance that I could help co-maintain this Flatpak? :-) I already maintain the Tremulous Flatpak on Flathub and Unvanquished is based on Tremulous.

**Comment by illwieckz** [comment]
@AsciiWolf the proposal is appreciated, though for the moment I'm not sure the amount of work required to update the flatpak is high enough to need third-party help. We definitely needed help to set it up at first, but now that it's done, it looks like updating on new release is basically a matter of a new git revision and some url to paste. Don't hesitate to submit pull requests if you seen any enhancement to do, that would be welcome !

**Pull request closed without merging**